packages/components/src/higher-order/with-context/test/index.js

```diff
@@ -39,6 +39,7 @@ describe( 'withContext', () => {
 		);
 
 		expect( wrapper.root.findByType( 'div' ).children[ 0 ] ).toBe( 'ok' );
+		expect( console ).toHaveWarned();
```

Comment:
Minor: Question came up from @danielbachhuber about testing with deprecated warnings, and potential for conflicts with its internal memoization. Made me think over the weekend that the fact we test against console logging is a bit of an abstraction piercing; while it seems obvious `deprecated` would log to console by its description, we can't know for certain, and seems we should test that `deprecated` was merely called, regardless of its internal implementation.

I'd like to think this would be as simple as? ...

```js
const deprecated = jest.mock( '@wordpress/deprecated' );

// ...

expect( deprecated ).toHaveBeenCalled();
```

Reply:
I think it should need:

```js
import deprecated from '@wordpress/deprecated';

jest.mock( '@wordpress/deprecated', () => jest.fn() );

expect( deprecated ).toHaveBeenCalled();
```

I can give it a try.

Reply:
Related docs: https://jestjs.io/docs/en/mock-functions#mocking-modules

Reply:
Refactored in https://github.com/WordPress/gutenberg/pull/8189/commits/cf41540bed800c32db0105cad762c92af9a57ed4.